### ISSUE TITLE
OpenAPI-Specification is invalid when using hierarchical pydantic model

### ISSUE THREAD

**Issue by swersig** [bentoml/BentoML#2570]
**Describe the bug**
If a hierarchical pydantic model is used, e.g.

```
from bentoml.io import JSON
from pydantic import BaseModel

class MyType(BaseModel):
    my_value: int
class MyRequest(BaseModel):
    my_type: MyType

@service.api(
    input=JSON(pydantic_model=MyRequest),
    output=JSON()
)
async def my_request(request):
    return 'my_response'
```

the OpenAPI specifiation generated by BentoML is invalid. It seems that BentoML reuses the schema generated by pydantic (which is a valid JSON schema) to describe the 'requestBody'. The pydantic-schema contains sub schemas for the sub types under the 'definitions' key. The OpenAPI-Spec only supports a subset of JSON-Schema that does not allow this 'inlining' of schemas. They should be placed under the global 'components/schemas' key.

**To Reproduce**
Steps to reproduce the issue:
1. Define a hierachical pydantic model
2. Use this model as input specifier for a service api
3. Open Swagger ui (e.g. http://localhost:3000)
4. Click on the service
5. Error message appears on top of page

Also you can not generate code with the openapi generator.

**Expected behavior**
1. The generated OpenAPI specification is valid. 
2. Code can be generated
3. No error appears in swagger ui

**Screenshots/Logs**
![error](https://user-images.githubusercontent.com/7602145/172607334-98aaa1b3-905f-4ee6-9094-b4883b0451db.jpg)

```
[ERROR]
org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 1, Warning count: 0
Errors:
        -attribute paths.'/my_request'(post).requestBody.content.'application/json'.schema.definitions is unexpected
```

**Environment:**
 - OS: Windows 10 x64
 - Python Version: Python 3.8.5
 - BentoML Version: 1.0.0a7, 1.0.0rc0

**Comment by swersig** [comment]
Hi @bojiang,

is this issue adressed in an upcoming release?

**Comment by parano** [comment]
Thanks for reporting the issue @swersig - our team is working on it and will be addressed in the next release.

see also: https://github.com/bentoml/BentoML/issues/2142

**Comment by aarnphm** [comment]
Development will be tracked at #2845 and will be included in 1.0.3 release.
